projects/admin/src/app/interceptor/user-current-library.interceptor.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * RERO ILS UI
+ * Copyright (C) 2023 RERO
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, version 3 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import {
+  HttpEvent,
+  HttpHandler,
+  HttpInterceptor,
+  HttpRequest
+} from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { UserService } from '@rero/shared';
+import { Observable } from 'rxjs';
+
+@Injectable()
+export class UserCurrentLibraryInterceptor implements HttpInterceptor {
+
+  /**
+   * Constructor
+   * @param _userService - UserService
+   */
+  constructor(private _userService: UserService) {}
+
+  /**
+   * Intercept http request
+   * Add current library of the logged user.
+   * @param request - HttpRequest
+   * @param next - HttpHandler
+   * @returns Handle of request
+   */
+  intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<unknown>> {
+    if (this._userService.user && this._userService.user.currentLibrary) {
+      request = request.clone({
```

Comment:
is it necessary to clone the request object (memory usage).
What's opinion of other devs (@jma, @rerowep, @lauren-d) about adding a `current_library` parameter on every HTTP call ?
Is it necessary to check than a `current_library` param already exists ?

Reply:
@zannkukai I discussed it with @jma. He's the one who told me to put it on the queries.

Reply:
@zannkukai in such case adding the `current_library` to all requests makes sense for me. For each request from the UI it can be interesting to transmit `the current_library`; the backend will decide if it is useful or not.

Reply:
@zannkukai For the `clone`, this seems to be good practice. There's an example in the Angular documentation: https://angular.io/guide/http-intercept-requests-and-responses#handle-interceptor-events